pallets/stellar-relay/src/lib.rs

```diff
@@ -565,27 +584,60 @@ pub mod pallet {
 
 				ensure!(node_id_found, Error::<T>::EnvelopeSignedByUnknownValidator);
```

Comment:
We need to be careful here, for some strange reason the SatoshiPay archive contains messages where one of the official SatoshiPay nodes is replaced with a completely different node. I am not sure why this is happening, maybe some misconfiguration in the archive or some historical deviation that might go back many years when we setup the archives for the first time?

See this example for ledger 46647587 from one of the SatoshiPay archives (all ledger entries look like this from that archive):

```
LEDGER 46647587
ScpQuorumSets
  Quorum Set
    threshold: 5
    inner quorum set:
        threshold: 2
        validators: BlockDaemon1 BlockDaemon2 BlockDaemon3 
    inner quorum set:
        threshold: 2
        validators: SDF3 SDF1 SDF2 
    inner quorum set:
        threshold: 2
        validators: Coinqvest2 Coinqvest3 Coinqvest1 
    inner quorum set:
        threshold: 2
        validators: SatoshiPay3 SatoshiPay2 SatoshiPay1 
    inner quorum set:
        threshold: 2
        validators: FranklinTempleton1 FranklinTempleton3 FranklinTempleton2 
    inner quorum set:
        threshold: 3
        validators: Lobster5 Lobster4 Lobster1 Lobster2 Lobster3 
    inner quorum set:
        threshold: 2
        validators: PublicNode1 PublicNode3 PublicNode2 
  Quorum Set
    threshold: 5
    inner quorum set:
        threshold: 2
        validators: BlockDaemon1 BlockDaemon2 BlockDaemon3 
    inner quorum set:
        threshold: 2
        validators: SDF3 SDF1 SDF2 
    inner quorum set:
        threshold: 2
        validators: Coinqvest2 Coinqvest3 Coinqvest1 
    inner quorum set:
        threshold: 2
        validators: FranklinTempleton1 FranklinTempleton3 FranklinTempleton2 
    inner quorum set:
        threshold: 3
        validators: Lobster5 Lobster4 Lobster1 Lobster2 Lobster3 
    inner quorum set:
        threshold: 2
        validators: SatoshiPay2 SatoshiPay1 GDGOUEQK5G6QINBH7277DC2MHVI4KXLAKEIB2OHN572AAONKLE5IMRRE 
    inner quorum set:
        threshold: 2
        validators: PublicNode1 PublicNode3 PublicNode2 
LedgerScpMessages (24)
  Prepare message from Lobster5
    quorumSet: quorumHash1
    b = (1, value1), p = (0, ⊥), p' = (0, ⊥), c.n = 0, h.n = 0
  Prepare message from BlockDaemon3
    quorumSet: quorumHash1
    b = (1, value1), p = (1, value1), p' = (0, ⊥), c.n = 0, h.n = 0
  Prepare message from PublicNode3
    quorumSet: quorumHash1
    b = (1, value1), p = (1, value1), p' = (0, ⊥), c.n = 0, h.n = 0
  Prepare message from FranklinTempleton3
    quorumSet: quorumHash1
    b = (1, value1), p = (1, value1), p' = (0, ⊥), c.n = 1, h.n = 1
  Prepare message from Lobster4
    quorumSet: quorumHash1
    b = (1, value1), p = (1, value1), p' = (0, ⊥), c.n = 1, h.n = 1
  Prepare message from Coinqvest2
    quorumSet: quorumHash1
    b = (1, value1), p = (1, value1), p' = (0, ⊥), c.n = 1, h.n = 1
  Prepare message from BlockDaemon2
    quorumSet: quorumHash1
    b = (1, value1), p = (1, value1), p' = (0, ⊥), c.n = 1, h.n = 1
  Prepare message from SatoshiPay2
    quorumSet: quorumHash1
    b = (1, value1), p = (1, value1), p' = (0, ⊥), c.n = 1, h.n = 1
  Prepare message from Lobster3
    quorumSet: quorumHash1
    b = (1, value1), p = (1, value1), p' = (0, ⊥), c.n = 1, h.n = 1
  Confirm message from BlockDaemon1
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from SDF3
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from SatoshiPay3
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from FranklinTempleton1
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from Coinqvest3
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from PublicNode1
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from SatoshiPay1
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from Lobster1
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from SDF1
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from SDF2
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from FranklinTempleton2
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from PublicNode2
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from Coinqvest1
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Confirm message from Lobster2
    quorumSet: quorumHash1
    b = (1, value1), p.n = 1, c.n = 1, h.n = 1
  Externalize message from GDGOUEQK5G6QINBH7277DC2MHVI4KXLAKEIB2OHN572AAONKLE5IMRRE
    quorumSet: quorumHash2
    c = (1, value1), h.n = 1
```

Reply:
>  I am not sure why this is happening, maybe some misconfiguration in the archive or some historical deviation that might go back many years when we set up the archives for the first time?

But the slot `46647587` [happened](https://stellar.expert/explorer/public/ledger/46647587) just about one week ago, right?
That's super weird. But I am honestly not too sure what we would do about it. We shouldn't really allow SCP messages signed by validators that are unknown to us and not in the tier 1 set. 

The 'unknown' node that externalized the message belongs to `quorumSet: quorumHash2` while all others belong to `quorumHash1`, what does it tell us? That it was not part of the tier 1 validator set? 

Oh but maybe it's only possible that the externalized message comes from a different node that is not in the tier 1 set. It was the same in the second snippet of your comment [here](https://github.com/pendulum-chain/spacewalk/pull/362#issuecomment-1591635924). It seems that in both cases, there are enough `Confirm` messages (ie. `14`) to build a valid quorum without considering the unknown node. So maybe the trick is not to insist on the externalized message originating from a tier 1 validator node, but also not count it in when checking the quorum validity when it's not.

Reply:
How would we loosen it up though? 

> It seems that in both cases, there are enough Confirm messages (ie. 14) to build a valid quorum without considering the unknown node. So maybe the trick is not to insist on the externalized message originating from a tier 1 validator node, but also not count it in when checking the quorum validity when it's not.

Do you think this could work? Or should we not even limit it to externalized messages coming from an 'unknown' validator but we basically allow any message to originate from an unknown validator but don't count it into the tier 1 quorum validity check?

Reply:
What is critical is that it seems to be always the `Externalize` message is the one that is sent from this replacement node. And the externalize message is of course the most critical one. I see only three options:
* fix our archives
* add these three replacements nodes to the list of "well-known nodes" (but that is a new notion that does not occur in the runtime yet)
* deep dive to understand how archives are composed.

Stellar now [also wants to improve this situation](https://github.com/stellar/stellar-core/issues/3769) after I inquired about the archive problem last week. But this can take a long time to fix.

I think the best approach is to fix our archives. Seems to be some misconfiguration.

Reply:
Okay, let's try to do option 1 then. Although I wouldn't really know how and where to start. Would we have to ask SDF devs about this again?

Reply:
I will reach out to people at Stellar.